plugins/catalog-backend/src/ingestion/LocationReaders.ts

```diff
@@ -248,4 +256,28 @@ export class LocationReaders implements LocationReader {
       }
     }
   }
+
+  private createProcessorContext(): ProcessorContext {
+    const handleLocation = this.handleLocation.bind(this);
+    return {
+      async read(location: LocationSpec): Promise<LocationProcessorResult> {
+        let locationResult: LocationProcessorResult | undefined;
+
+        await handleLocation(
```

Comment:
Feels a bit hacky x), imo we should look into separating data fetching from locations out into its own thing. Separate PR either way though.

Regarding having this in context I'm not sure that's quite what @freben meant by context, where he was more talking about a place for processors to store data. This could be something that is passed to processors to use, just like `emit` etc.

Reply:
You mean, instead of passing the context object, just passt the callback? The context object feels quite empty anyways 😁

Reply:
Yeah have the call be

```ts
processor.readLocation({
  location: item.location,
  optional: item.optional,
  emit,
  read,
})
```